src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs

```diff
@@ -799,6 +794,12 @@ public virtual NotFoundObjectResult NotFound(object value)
             => new NotFoundObjectResult(value);
 
         /// <summary>
+        /// Creates a <see cref="PageResult"/> object that renders the page.
+        /// </summary>
+        /// <returns>The <see cref="PageResult"/>.</returns>
+        public virtual PageResult Page() => new PageResult(PageContext.Page, this);
```

Comment:
Another alternative might be to rename this method to `CurrentPage()`. It seems a bit clearer too in the intent of the method?